### PR TITLE
KONFLUX-191: Prow configuration for cert-manager-operator-release repo

### PR DESCRIPTION
**Pull request by bharath-b-rh**
Add Prow configuration code for new cert-manager-operator-release repository which hosts configurations required for konflux builds.

### CODE REVIEW

File: core-services/prow/02_config/openshift/cert-manager-operator-release/_prowconfig.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+branch-protection:
+  orgs:
+    openshift:
+      repos:
```
[the remark this answers is not in the document]
Comment (swghosh) [reply core-services/prow/02_config/openshift/cert-manager-operator-release/_prowconfig.yaml:4]: https://docs.ci.openshift.org/docs/architecture/branch-protection/#is-it-possible-to-disable-the-branch-protection-for-my-repository-or-require-jobs-that-are-not-managed-by-prow
allows making the Konflux CI jobs as Required before a PR can merge.

File: core-services/prow/02_config/openshift/cert-manager-operator-release/_prowconfig.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            release-1.15:
+              protect: true
+            main:
+              protect: true
+tide:
+  queries:
+    labels:
+    - approved
+    - lgtm
+    - jira/valid-bug
```
[the remark this answers is not in the document]
Comment (bharath-b-rh) [reply core-services/prow/02_config/openshift/cert-manager-operator-release/_prowconfig.yaml:18]: Saw it now, should have been `jira/valid-reference`

File: core-services/prow/02_config/openshift/cert-manager-operator-release/_prowconfig.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            release-1.15:
+              protect: true
+            main:
+              protect: true
+tide:
+  queries:
+  - labels:
+    - approved
+    - lgtm
+    - jira/valid-bug
```
Comment (bharath-b-rh) [inline core-services/prow/02_config/openshift/cert-manager-operator-release/_prowconfig.yaml:19]:
```suggestion
    - jira/valid-reference
```

File: ci-operator/config/openshift/cert-manager-operator-release/openshift-cert-manager-operator-release-main.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.22
```
[the remark this answers is not in the document]
Comment (bharath-b-rh) [reply ci-operator/config/openshift/cert-manager-operator-release/openshift-cert-manager-operator-release-main.yaml:5]: This was auto-generated.

File: ci-operator/config/openshift/cert-manager-operator-release/openshift-cert-manager-operator-release-main.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+releases:
+  initial:
+    integration:
+      name: "4.18"
+      namespace: ocp
+  latest:
+    integration:
+      include_built_images: true
+      name: "4.18"
+      namespace: ocp
```
[the remark this answers is not in the document]
Comment (bharath-b-rh) [reply ci-operator/config/openshift/cert-manager-operator-release/openshift-cert-manager-operator-release-main.yaml:15]: This was auto-generated.

File: ci-operator/config/openshift/cert-manager-operator-release/openshift-cert-manager-operator-release-main.yaml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  '*':
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: verify
+  commands: |
+    make verify
```
[the remark this answers is not in the document]
Comment (bharath-b-rh) [reply ci-operator/config/openshift/cert-manager-operator-release/openshift-cert-manager-operator-release-main.yaml:26]: No, docker is set as default.

File: core-services/prow/02_config/openshift/cert-manager-operator-release/_pluginconfig.yaml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+approve:
+- commandHelpLink: ""
+  repos:
+  - openshift/cert-manager-operator-release
+  require_self_approval: false
+lgtm:
+- repos:
+  - openshift/cert-manager-operator-release
+  review_acts_as_lgtm: true
+plugins:
```
Comment (swghosh) [inline core-services/prow/02_config/openshift/cert-manager-operator-release/_pluginconfig.yaml:10]:
```suggestion
triggers:
- repos:
  - openshift/cert-manager-operator-release
  trusted_apps:
  - red-hat-konflux
plugins:
```
Just a random suggestion, with this you'd still have to check if adding this whitelists the Konflux bot for PR tests and we don't have to add the `/ok-to-test` label each time manually!

File: core-services/prow/02_config/openshift/cert-manager-operator-release/_prowconfig.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (swghosh) [reply core-services/prow/02_config/openshift/cert-manager-operator-release/_prowconfig.yaml:4]: Looks like you'd need:

eg.
```
branches:
    release-1.15:
        protect: true
            required_status_checks:
                contexts:
                - Red Hat Konflux
     release-1.14:
        protect: true
            required_status_checks:
                contexts:
                - Red Hat Konflux
```
to get "Required" for those Konflux jobs.

File: ci-operator/config/openshift/cert-manager-operator-release/openshift-cert-manager-operator-release-main.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (bharath-b-rh) [reply ci-operator/config/openshift/cert-manager-operator-release/openshift-cert-manager-operator-release-main.yaml]: suggestion incorporated.

File: ci-operator/config/openshift/cert-manager-operator-release/openshift-cert-manager-operator-release-main.yaml [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  container:
+    from: src
+zz_generated_metadata:
+  branch: main
```
[the remark this answers is not in the document]
Comment (bharath-b-rh) [reply ci-operator/config/openshift/cert-manager-operator-release/openshift-cert-manager-operator-release-main.yaml:22]: suggestion incorporated.

File: core-services/prow/02_config/openshift/cert-manager-operator-release/_pluginconfig.yaml (hunk 7 above)
Comment (bharath-b-rh) [reply core-services/prow/02_config/openshift/cert-manager-operator-release/_pluginconfig.yaml:15]: suggestion incorporated.

**Comment by bharath-b-rh** [comment]
/pj-rehearse pull-ci-openshift-cert-manager-operator-release-main-verify pull-ci-openshift-cert-manager-operator-release-release-1.14-verify pull-ci-openshift-cert-manager-operator-release-release-1.15-verify

**Comment by bharath-b-rh** [comment]
/pj-rehearse pull-ci-openshift-cert-manager-operator-release-main-verify pull-ci-openshift-cert-manager-operator-release-release-1.14-verify pull-ci-openshift-cert-manager-operator-release-release-1.15-verify

**Comment by bharath-b-rh** [comment]
/pj-rehearse pull-ci-openshift-cert-manager-operator-release-main-verify pull-ci-openshift-cert-manager-operator-release-release-1.14-verify pull-ci-openshift-cert-manager-operator-release-release-1.15-verify

**Comment by bharath-b-rh** [comment]
/pj-rehearse pull-ci-openshift-cert-manager-operator-release-main-verify pull-ci-openshift-cert-manager-operator-release-release-1.14-verify pull-ci-openshift-cert-manager-operator-release-release-1.15-verify

**Comment by bharath-b-rh** [comment]
/pj-rehearse pull-ci-openshift-cert-manager-operator-release-main-verify pull-ci-openshift-cert-manager-operator-release-release-1.14-verify pull-ci-openshift-cert-manager-operator-release-release-1.15-verify

**Comment by bharath-b-rh** [comment]
/pj-rehearse pull-ci-openshift-cert-manager-operator-release-main-verify pull-ci-openshift-cert-manager-operator-release-release-1.14-verify pull-ci-openshift-cert-manager-operator-release-release-1.15-verify

**Comment by bharath-b-rh** [comment]
/pj-rehearse pull-ci-openshift-cert-manager-operator-release-main-verify pull-ci-openshift-cert-manager-operator-release-release-1.14-verify pull-ci-openshift-cert-manager-operator-release-release-1.15-verify

**Pull request merged**